contracts/Vault.sol

```diff
@@ -224,12 +225,36 @@ contract Vault is
         claimableYield = sharesAmount - perfFee;
     }
 
+    /// @inheritdoc IVault
+    function depositForGroupId(uint256 _groupId, DepositParams calldata _params)
+        external
+        nonReentrant
+        whenNotPaused
+        returns (uint256[] memory depositIds)
+    {
+        if (depositGroupIdOwner[_groupId] != msg.sender)
+            revert VaultSenderNotOwnerOfGroupId();
+
+        depositIds = _doDeposit(_groupId, _params);
+    }
+
     /// @inheritdoc IVault
     function deposit(DepositParams calldata _params)
         external
         nonReentrant
         whenNotPaused
         returns (uint256[] memory depositIds)
+    {
+        depositGroupIdOwner[_depositGroupIds] = msg.sender;
+
+        depositIds = _doDeposit(_depositGroupIds, _params);
+
+        _depositGroupIds++;
```

Comment:
i think `++_depositGroupIds` or `_depositGroupIds += 1` will save us a bit of gas here

Reply:
![image](https://user-images.githubusercontent.com/4612986/166911697-3ec8956b-1e15-4765-b833-3c2b57ce5dae.png)

literally a 5 gas difference :joy: but I guess every little bit can help